graphql/websocket.go

```diff
@@ -0,0 +1,186 @@
+package graphql
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	webSocketMethod         = "websocket"
+	webSocketTypeConnInit   = "connection_init"
+	webSocketTypeConnAck    = "connection_ack"
+	webSocketTypeSubscribe  = "subscribe"
+	webSocketTypeNext       = "next"
+	webSocketTypeError      = "error"
+	webSocketTypeComplete   = "complete"
+	websocketConnAckTimeOut = time.Second * 30
```

Comment:
silly question, are these GraphQL things or websocket things or what? is there some spec or documentation we can link to that lists them? (are they things a client might need to configure in the future?)

Reply:
```
        webSocketMethod         = "websocket"
```
this is just an identifier I chose to identify the websocket method from the `http.MethodGet` and `http.MethodPost`

```
	webSocketTypeConnInit   = "connection_init"
	webSocketTypeConnAck    = "connection_ack"
	webSocketTypeSubscribe  = "subscribe"
	webSocketTypeNext       = "next"
	webSocketTypeError      = "error"
	webSocketTypeComplete   = "complete"
```
these are graphQL constants, they are defined [here](https://github.com/99designs/gqlgen/blob/fceb33111bef2bbbf14468d78888098ac8968351/graphql/handler/transport/websocket_graphql_transport_ws.go#L14)

```
        websocketConnAckTimeOut = time.Second * 30
```
this is arbitrary, it could be added to a webSocket client configuration